### PR TITLE
TPU Provisioner: JobSet related fixes

### PR DESCRIPTION
**Pull request by nstogner**
* `go mod tidy`
* Add missing JobSet RBAC
* Add missing JobSet scheme

### CODE REVIEW

File: tpu-provisioner/go.mod [hunk 1]
```diff
@@ -1,9 +1,10 @@
 module github.com/GoogleCloudPlatform/ai-on-gke/tpu-provisioner
 
-go 1.22
+go 1.22.0
```
Comment (danielvegamyhre) [inline tpu-provisioner/go.mod:3]: Did you test building the image after this change? I had to remove the ".0" otherwise the image wouldn't build.

Comment (nstogner) [reply tpu-provisioner/go.mod:3]: Yes - retesting now though - PS: this was a change that was made by `go` itself

Comment (nstogner) [reply tpu-provisioner/go.mod:3]: Confirmed a `podman build` works for me

**Review by danielvegamyhre — APPROVED** [review]

**Pull request merged**